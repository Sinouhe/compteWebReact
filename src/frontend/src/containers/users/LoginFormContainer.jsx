import React from 'react';
import LoginForm from '../../form/loginForm/LoginForm';
import { signinUser } from '../../actions/users';
import { connect } from 'react-redux';
import { Card } from 'antd';
import './users.css';

class LoginFormContainer extends React.Component {

    handleSubmit = (credentials, setError) => {
        this.props.signinUser(credentials, setError);
    }

    render() {
        return (
            <Card title="Login" style={{ 
                                        borderRadius: 8, 
                                        marginLeft: '6%', 
                                        top: 20
                                        }} className='card-login-form'>
                <LoginForm handleSubmitParent={this.handleSubmit}/>
            </Card>
        );
    }
}

const mapDispatchToProps = {
    signinUser
};
  
export default connect(null, mapDispatchToProps)(LoginFormContainer);
