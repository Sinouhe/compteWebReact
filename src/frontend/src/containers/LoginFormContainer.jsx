import React, { Fragment } from 'react';
import LoginForm from '../form/loginForm/LoginForm';
import { signinUser } from '../actions/users';
import { connect } from 'react-redux';

class LoginFormContainer extends React.Component {

    handleSubmit = (credentials) => {
        this.props.signinUser(credentials);
    }

    render() {
        return (
            <Fragment>
                <LoginForm handleSubmitParent={this.handleSubmit}/>
            </Fragment>
        );
    }
}

const mapDispatchToProps = {
    signinUser
};
  
export default connect(null, mapDispatchToProps)(LoginFormContainer);
