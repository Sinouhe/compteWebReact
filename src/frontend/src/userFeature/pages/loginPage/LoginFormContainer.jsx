import React from 'react';
import LoginForm from '../../form/loginForm/LoginForm';
import { signinUserAction } from '../../actions/users';
import { connect } from 'react-redux';
import { Card } from 'antd';
import { withRouter } from 'react-router-dom';

class LoginFormContainer extends React.Component {

    componentDidMount = () => {
        if(this.props.isLoggedIn === true) {
            this.props.history.push('/');
        }
    }

    handleSubmit = (credentials, setError) => {
        this.props.signinUserAction(credentials, setError, this.props.history);
    }

    render() {
        return ( 
            <Card title="Login" style={{ 
                                        borderRadius: 8, 
                                        marginLeft: '6%', 
                                        top: 20
                                        }} className='card-login-form'>
                                            <div>salut</div>
                <LoginForm handleSubmitParent={this.handleSubmit}/>
            </Card>
        );
    }
}

const mapStateToProps = state => {
    console.log(state)
    return {
      isLoggedIn: state.users.isLoggedIn
    };
};

const mapDispatchToProps = {
    signinUserAction
};
  
const LoginFormContainerWithRouter = withRouter(connect(mapStateToProps, mapDispatchToProps)(LoginFormContainer));
const LoginFormContainerConnected = connect(mapStateToProps, mapDispatchToProps)(LoginFormContainer);

export {
    LoginFormContainerWithRouter,
    LoginFormContainerConnected,
    LoginFormContainer
};