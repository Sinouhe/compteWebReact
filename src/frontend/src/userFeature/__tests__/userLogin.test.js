import React from 'react';
import { LoginFormContainer } from '../pages/loginPage/LoginFormContainer';
import {shallow} from 'enzyme';
import json from 'enzyme-to-json';


describe('My Connected React-Redux Component', () => {
    let wrapper;

    beforeEach(() => {  
      wrapper = shallow(
        <LoginFormContainer />
      );
    });    

  it('should render with given state from Redux store', () => {
        expect(json(wrapper)).toMatchSnapshot();
    });

    /*
    it('should dispatch an action on submit form', () => {
      renderer.act(() => {
        wrapper.root.find( (el) => el.type == 'button' && el.props.id == 'loginFormSubmitButton').props.onClick();
      });
      expect(mockStoreInitialized.dispatch).toHaveBeenCalledTimes(1);
      expect(store.dispatch).toHaveBeenCalledWith(
        myAction({ payload: 'sample text' })
      );
    });
    */

   
  });