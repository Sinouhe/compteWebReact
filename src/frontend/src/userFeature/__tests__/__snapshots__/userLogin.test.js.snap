// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<Card
  className="card-login-form"
  style={
    Object {
      "borderRadius": 8,
      "marginLeft": "6%",
      "top": 20,
    }
  }
  title="Login"
>
  <div>
    salut
  </div>
  <LoginForm
    handleSubmitParent={[Function]}
  />
</Card>
`;
